meshnet/server.go

```diff
@@ -1369,6 +1374,125 @@ func (s *Server) RenamePeer(
 	}, nil
 }
 
+func (s *Server) ChangeMachineNickname(
+	ctx context.Context,
+	req *pb.ChangeMachineNicknameRequest,
+) (*pb.RenamePeerResponse, error) {
+	if !s.ac.IsLoggedIn() {
+		return &pb.RenamePeerResponse{
+			Response: &pb.RenamePeerResponse_ServiceErrorCode{
+				ServiceErrorCode: pb.ServiceErrorCode_NOT_LOGGED_IN,
+			},
+		}, nil
+	}
+
+	var cfg config.Config
+	if err := s.cm.Load(&cfg); err != nil {
+		s.pub.Publish(err)
+		return &pb.RenamePeerResponse{
+			Response: &pb.RenamePeerResponse_ServiceErrorCode{
+				ServiceErrorCode: pb.ServiceErrorCode_CONFIG_FAILURE,
+			},
+		}, nil
+	}
+
+	// check if meshnet is enabled
+	if !cfg.Mesh {
+		return &pb.RenamePeerResponse{
+			Response: &pb.RenamePeerResponse_MeshnetErrorCode{
+				MeshnetErrorCode: pb.MeshnetErrorCode_NOT_ENABLED,
+			},
+		}, nil
+	}
+
+	token := cfg.TokensData[cfg.AutoConnectData.ID].Token
+
+	// check info and re-register if needed
+	if !s.mc.IsRegistrationInfoCorrect() {
+		return &pb.RenamePeerResponse{
+			Response: &pb.RenamePeerResponse_ServiceErrorCode{
+				ServiceErrorCode: pb.ServiceErrorCode_CONFIG_FAILURE,
+			},
+		}, nil
+	}
+
+	if req.Nickname == "" {
+		if cfg.MeshDevice.Nickname == "" {
```

Comment:
Why need this check? Plz add comment in the code. Probably it is when trying to remove nickname when it was not set. But in that case maybe we should say to user exactly that?

Reply:
Ok, I'll return an error also for this case.